### PR TITLE
Display PDF document with lib

### PR DESCRIPTION
**Pull request by enieber**
- [x] translate lib to en
- [x] upload lib to @libresign/vue-pdf-editor
- [x] remove files not use
- [x] render pdf with lib

### CODE REVIEW

**Review by giovannism20 — APPROVED** [review]

File: src/views/Request.vue [hunk 1]
```diff
@@ -5,18 +5,22 @@
...
 					status="0"
 					status-text="none"
 					@sidebar="showSidebar = true" />
-				<NcButton @click="showModalUploadFromUrl">
-					{{ t('libresign', 'Upload from URL') }}
-					<template #icon>
-						<LinkIcon :size="20" />
-					</template>
-				</NcButton>
+				<!-- <NcButton @click="showModalUploadFromUrl"> -->
```
Comment (vitormattos) [inline src/views/Request.vue:18]: Maybe would be best to hide this item:

```suggestion
				<NcButton @click="showModalUploadFromUrl" v-if="false">
```

Will be less changes and the linter don't will cry

File: src/views/Request.vue [hunk 2]
```diff
@@ -54,7 +59,9 @@
 		<LibresignTab v-if="showSidebar"
 			:prop-file="file"
 			:prop-name="file.name"
-			@close="showSidebar = false" />
+			@close="showSidebar = false"
```
Comment (vitormattos) [inline src/views/Request.vue:62]: necessary fix the linter

run the follow commands to fix this check and others in this PR:
```
npm run lint:fix
npm run stylelint:fix
```

**Review by vitormattos — CHANGES_REQUESTED** [review]

**Review by vitormattos — COMMENTED** [review]
I know that maybe will sound strange put all of this files in a same directory, but since that this is a package that we are putting inside the project to be more fast, maybe will be best to have a folder like `src/pdf-editor` and inside this folder put all files of this package. Making this will be more easy to extract the package in another moment.

**Comment by vitormattos** [comment]
Will be necessary run a rebase and also remove the `package-lock.json`, run `npm i` to generate a file without conflicts and version the changes of `package-lock-json`.

**Comment by vitormattos** [comment]
The DCO disliked your commit.

Have a very simple way to solve the problems of forgotten sign-off. You can add a commit template.

Steps to do this:
- Create the file that will contains your template, i.e. `~/.git_commit_template`
- Add the follow content to your file:
  ```


  Signed-off-by: Your Name <your-name@the.domain>
  ```
  > Pay **attention** to add two rows before the signature
- Configure this file to be your commit template:
  ```
  git config commit.template ~/.git_commit_template
  ```

If you prefer to do this to all of your commits, you can do the same config with global scope:
```
git config --global commit.template ~/.git_commit_template
```

**Comment by vitormattos** [comment]
Ref: https://github.com/LibreSign/libresign/issues/1972

**Comment by enieber** [comment]
The problem to load xcc-vue-pdf 

image

**Comment by vitormattos** [comment]
Sounds that is this a cache of previous build because I didn't found the place that this file is generated.
By the way, did you opened this js file in another tab to check if the response is a JS content?

**Comment by vitormattos** [comment]
The conflict with package files will be solved by a rebase from main branch.

File: package.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
   },
   "dependencies": {
     "@fontsource/dancing-script": "^5.0.15",
+    "@libresign/vue-pdf-editor": "^1.0.0",
```
Comment (vitormattos) [inline package.json:21]: Will we use this package or the package `xcc-pdf-editor` ?

About the other changes, will we really need to add all of these packages? Looking the .vue files I didn't found reference to all of the follow packages that was added to package.json

**Comment by enieber** [comment]
I change all reference xcc-pdf-editor to vue-pdf-editor but error not updated, maybe this error is not in lib

image

File: package.json (hunk 3 above)
Comment (enieber) [reply package.json:21]: the repository of lib code.

https://github.com/LibreSign/vue-pdf-editor/

**Comment by vitormattos** [comment]
need to check the linter.

At package.json have a script block with commands that will be good to do this.

File: src/views/RequestDocument.vue [hunk 4]
```diff
@@ -0,0 +1,295 @@
+<template>
```
Comment (vitormattos) [inline src/views/RequestDocument.vue:1]: Sounds that this change is out of context of this PR.

**Review by vitormattos — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 5]
```diff
@@ -12,3 +12,4 @@ node_modules/
 .phpunit.result.cache
 *.phar
 /src/__test__/coverage
+.tgz
```
[the remark this answers is not in the document]
Comment (enieber) [reply .gitignore:15]: not, is required to not version local pack

File: src/Components/Request/VisibleElements.vue [hunk 6]
```diff
@@ -285,11 +257,16 @@ export default {
 		},
 		async loadDocument() {
 			try {
+				this.loading = true
 				this.signers = []
-				this.document = await axios.get(generateOcsUrl(`/apps/libresign/api/v1/file/validate/file_id/${this.file.nodeId}`))
-				this.document = this.document.data
+				const urlFile = generateOcsUrl(`/apps/libresign/api/v1/file/validate/file_id/${this.file.nodeId}`)
```
[the remark this answers is not in the document]
Comment (enieber) [reply src/Components/Request/VisibleElements.vue:262]: is better user js pattern and not need learn about how function generateOcsUrl work internal.

File: src/main.js [hunk 7]
```diff
@@ -26,6 +26,7 @@ import { generateFilePath } from '@nextcloud/router'
 import { getRequestToken } from '@nextcloud/auth'
 import { sync } from 'vuex-router-sync'
 import { translate, translatePlural } from '@nextcloud/l10n'
+import VuePdfEditor from '@libresign/vue-pdf-editor'
```
[the remark this answers is not in the document]
Comment (enieber) [reply src/main.js:29]: this lib is required include global

File: src/views/RequestDocument.vue (hunk 4 above)
Comment (enieber) [reply src/views/RequestDocument.vue:1]: fix lint error

File: .gitignore (hunk 5 above)
[the remark this answers is not in the document]
Comment (vitormattos) [reply .gitignore:15]: But, haven't a place generating this file and the LibreSign don't pack a tgz file.

I think that this was necessary only to make a test at development environment, not?

File: src/Components/Request/VisibleElements.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (vitormattos) [reply src/Components/Request/VisibleElements.vue:262]: Another good way is to indent the code:
```suggestion
				const urlFile = generateOcsUrl('/apps/libresign/api/v1/file/validate/file_id/{nodeId}', {
				    nodeId: ${this.file.nodeId}
				})
```

Follow the documentation of this function:
https://nextcloud-libraries.github.io/nextcloud-router/functions/generateOcsUrl.html

File: src/main.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (vitormattos) [reply src/main.js:29]: This isn't good. What we need to don't make this?

File: src/views/RequestDocument.vue (hunk 4 above)
Comment (vitormattos) [reply src/views/RequestDocument.vue:1]: This is a new file, isn't a lint fix.

File: src/main.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (enieber) [reply src/main.js:29]: I dont know.

**Comment by vitormattos** [comment]
I will do a rebase to solve the conflicts

**Pull request merged**